### DISCUSSION TITLE
Github Actions: Repository name must be lowercase

### DISCUSSION THREAD

**Discussion by DenizUgur** [community/community#27086] [Actions]
<p>My username contains uppercase letters and this hinders my ability to push to <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a>. Are there any workarounds for this? I just want to push to the registry and see it on my GitHub Repository.</p>
<pre><code class="lang-auto">error: invalid tag "ghcr.io/DenizUgur/safari:b110b2ab2ad69f9fe64380feda88a7e00a12e88b": repository name must be lowercase
</code></pre>
<p><a href="https://github.com/DenizUgur/SAFARI/blob/b110b2ab2ad69f9fe64380feda88a7e00a12e88b/.github/workflows/build.yml" rel="noopener nofollow ugc">workflow file</a></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Image names must be lower case, try using the lower case version of your username. You can create an environment variable with the lower case version, so you don’t have to hardcode it. Looking at your workflow I guess <code>github.repository_owner</code> is the issue, so you could do this:</p>
<pre><code class="lang-yaml">- name: set lower case owner name
  run: |
    echo "OWNER_LC=${OWNER,,}" &gt;&gt;${GITHUB_ENV}
  env:
    OWNER: '${{ github.repository_owner }}'
</code></pre>
<p>And then use the <code>OWNER_LC</code> environment variable instead of <code>github.repository_owner</code>.</p>

**Reply by A7mmed1** [reply]
worked for me, thanks.

**Reply by lokeshkato** [reply]
It didnt work for me 😢  I get a 403 in response

```
- name: Set lowercase image name
   run: |
     echo "IMAGE_NAME=ghcr.io/${GITHUB_REPOSITORY_OWNER,,}/xyz" >> $GITHUB_ENV
     echo "TAG=latest" >> $GITHUB_ENV
```

Any suggestions

**Reply by jsoref** [reply]
@lokeshkato: open a new discussion with a link to your workflow/action run.

**Comment by DenizUgur** [comment]
<p>Will use, thank you. I thought GitHub usernames were case-sensitive.</p>

**Comment by jsoref** [comment]
<p>I think GitHub is case-preserving as opposed to case-sensitive.<br>
You could technically try to rename your username (probably to a placeholder and then to the lowercase form).</p>
<p>I’m not sure I’d bother.</p>

**Comment by airtower-luna** [comment]
<p>It’s a general Docker limitation, not specific to GHCR.</p>

**Comment by clarkbw** [comment]
<p>If you use our <a href="https://github.com/actions/starter-workflows/blob/82c43562976aef3d8f02643e00e979d65993244c/ci/docker-publish.yml">docker-publish default workflow</a> you’ll get a lot of this for free. The metadata action manages the lowercasing of names and extracts the right extra data for you.</p>

**Comment by matriculus** [comment]
What about organisation name? docker wont accept upper case letters to build image, github wont accept wrong case for pushing image. Is there a solution for this case?

**Reply by jsoref** [reply]
- There's no distinction between a user and an organization from the perspective of this api: https://github.com/orgs/community/discussions/27086#discussioncomment-3254548
